lensterxyz/lenster
# feat: add image attachment support to chat

This PR adds support for sending E2E encrypted images in messages.

It uses the newly introduced `Attachment` and `RemoteAttachment` content types which you can read more about here: https://xmtp.org/blog/attachments-and-remote-attachments.

https://user-images.githubusercontent.com/483/227057859-b5dd6059-cdd5-4224-81ac-22d97029afe9.mov

I had to change some of the hardcoded height values that were in the messages area before to accommodate the composer growing when you're about to send an image. I also added a slight background color to the composer because it looked confusing to have a bunch of white space when you had an attachment preview in there. Feel free to change any of that if it doesn't work for you.

## Type of change


## How should this be tested?

You should be able to add an image by clicking the photo button in the chat view.
